src/rust/ide/src/controller/graph.rs

```diff
@@ -194,17 +191,16 @@ impl Handle {
     }
 
     /// Retrieves metadata for the given node.
-    pub fn node_metadata(&self, _id:ast::ID) -> FallibleResult<NodeMetadata> {
-        // todo!()
-        #[derive(Clone,Copy,Debug,Display,Fail)]
-        struct NotImplemented;
-        Err(NotImplemented.into())
+    pub fn node_metadata(&self, id:ast::ID) -> FallibleResult<NodeMetadata> {
+        self.module().node_metadata(id)
     }
 
-    /// Update metadata for the given node.
-    pub fn update_node_metadata<F>(&self, _id:ast::ID, _updater:F) -> FallibleResult<NodeMetadata>
+    /// Modify metadata of given node.
+    /// If ID doesn't have metadata, empty (default) metadata is inserted.
+    pub fn with_node_metadata<F>(&self, id:ast::ID, updater:F) -> FallibleResult<NodeMetadata>
     where F : FnOnce(&mut NodeMetadata) {
-        todo!()
+        self.module().with_node_metadata(id, updater);
+        self.node_metadata(id)
```

Comment:
I'm confused why this method returns the metadata, while the module's one doesn't. I think they should either behave the same or be named differently.